test/suite/test_hs11.py

```diff
@@ -139,44 +139,44 @@ def test_ts_updates_donot_clears_hs(self):
-                cursor[self.create_key(i)] = value1
-                self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(ts))
+                for retry in self.retry():
+                    with retry.transaction(commit_timestamp = ts):
+                        cursor[self.create_key(i)] = value1
 
         # Reconcile and flush versions 1-3 to the history store.
         self.session.checkpoint()
 
         # Remove the key with timestamp 10.
         for i in range(1, self.nrows):
             if i % 2 == 0:
-                self.session.begin_transaction()
-                cursor.set_key(self.create_key(i))
-                cursor.remove()
-                self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(10))
+                for retry in self.retry():
+                    with retry.transaction(commit_timestamp = 10):
+                        cursor.set_key(self.create_key(i))
+                        cursor.remove()
 
         # Reconcile and remove the obsolete entries.
         self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(10))
         self.session.checkpoint()
 
         # Now apply an update at timestamp 20.
         for i in range(1, self.nrows):
-            self.session.begin_transaction()
-            cursor[self.create_key(i)] = value2
-            self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(20))
+                for retry in self.retry():
+                    with retry.transaction(commit_timestamp = 20):
+                        cursor[self.create_key(i)] = value2
 
         # Ensure that we didn't select old history store content even if it is not blew away.
-        self.session.begin_transaction('read_timestamp=' + self.timestamp_str(10))
-        for i in range(1, self.nrows):
-            if i % 2 == 0:
-                cursor.set_key(self.create_key(i))
-                if self.value_format == '8t':
-                    self.assertEqual(cursor.search(), 0)
-                    self.assertEqual(cursor.get_value(), 0)
-                else:
-                    self.assertEqual(cursor.search(), wiredtiger.WT_NOTFOUND)
-            else:
-                self.assertEqual(cursor[self.create_key(i)], value1)
-        self.session.rollback_transaction()
+        for retry in self.retry():
+            with retry.transaction(read_timestamp = 10):
+                for i in range(1, self.nrows):
+                    if i % 2 == 0:
+                        cursor.set_key(self.create_key(i))
+                        if self.value_format == '8t':
+                            self.assertEqual(cursor.search(), 0)
+                            self.assertEqual(cursor.get_value(), 0)
+                        else:
+                            self.assertEqual(cursor.search(), wiredtiger.WT_NOTFOUND)
+                    else:
+                        self.assertEqual(cursor[self.create_key(i)], value1)
```

Comment:
Have we lost the testing of `rollback_transaction` here? Ok - I see that the `transaction` method rolls back if `read_timestamp` is set. That works for these specific cases but is subtle and won't provide the same testing for all tests (like `test_timestamp02.py` which uses `read_timestamp+commit_transaction` in its `check` function.

Maybe for this one add a comment here and all the other places that states the assumption 'FIXME-WT-XXXX Using read_timestamp means retry.transaction will rollback the transaction' for now. I think this will need to be fixed with whatever real, general mechanism is used. This fixes the ci-blocker, but is not a forever solution.

Reply:
Would it help to have an explicit arg to force the rollback?  Like:
```
            for retry in self.retry():
                with retry.transaction(read_timestamp = ts, rollback_readonly_transaction = True):
                    ....
```
I don't _think_ the rollback (vs commit) is an important part of the test, it is a design pattern used since the transaction is readonly.

Reply:
How about just `rollback=True` since there might be (later, other) cases where we rollback updates for other testing purposes?

I believe these tests are written this way because MongoDB uses rollback on read-only transactions. So it more closely tests/tracks how the server is using WT.